serialisation/yaml/XPathExtractorParser_test.go

```diff
@@ -0,0 +1,35 @@
+package yaml
+
+import (
+	"github.com/guzzlerio/corcel/core"
+	"github.com/guzzlerio/corcel/extractors"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+)
+
+var _ = Describe("XPathExtractorParser", func() {
+	It("Parses", func() {
```

Comment:
Done and changed the signature of the interface to also return an error.  Added failure mode tests to all the extractors